src/aiai_eval/task.py

```diff
@@ -54,7 +82,567 @@ def evaluate(self, model_id: str) -> Dict[str, dict]:
+            else:
+                raise ValueError(f"The supertask `{supertask}` was not recognised.")
+
+            model = model_cls.from_pretrained(
+                model_config.model_id,
+                revision=model_config.revision,
+                use_auth_token=self.evaluation_config.use_auth_token,
+                config=config,
+                cache_dir=self.evaluation_config.cache_dir,
+                from_flax=from_flax,
+            )
+
+        except (OSError, ValueError):
+            msg = (
+                f"The model {model_config.model_id} either does not have a "
+                "frameworks registered, or it is a private model. If it is a "
+                "private model then enable the `--use-auth-token` flag and "
+                "make  sure that you are logged in to the Hub via the "
+                "`huggingface-cli login` command."
+            )
+            raise InvalidEvaluation(msg)
+
+        # Ensure that the labels of the model are consistent with the labels of the
+        # dataset
+        self._adjust_label_ids(model=model, model_config=model_config)
+
+        # If the model is a subclass of a RoBERTa model then we have to add a prefix
+        # space to the tokens, by the way the model is constructed.
+        m_id = model_config.model_id
+        prefix = "Roberta" in type(model).__name__
+        params = dict(use_fast=True, add_prefix_space=prefix)
+        tokenizer = AutoTokenizer.from_pretrained(
+            m_id,
+            revision=model_config.revision,
+            use_auth_token=self.evaluation_config.use_auth_token,
+            **params,
+        )
+
+        # Set the maximal length of the tokenizer to the model's maximal length.
+        # This is required for proper truncation
+        if (
+            not hasattr(tokenizer, "model_max_length")
+            or tokenizer.model_max_length > 1_000
+        ):
+
+            if hasattr(tokenizer, "max_model_input_sizes"):
+                all_max_lengths = tokenizer.max_model_input_sizes.values()
+                if len(list(all_max_lengths)) > 0:
+                    min_max_length = min(list(all_max_lengths))
+                    tokenizer.model_max_length = min_max_length
+                else:
+                    tokenizer.model_max_length = 512
+            else:
+                tokenizer.model_max_length = 512
+
+        return dict(model=model, tokenizer=tokenizer)
+
+    def _load_spacy_model(self, model_config: ModelConfig) -> Dict[str, Any]:
+        """Load a spaCy model.
+
+        Args:
+            model_config (ModelConfig):
+                The configuration of the model.
+
+        Returns:
+            dict:
+                A dictionary containing at least the key 'model', with the value being
+                the model. Can contain other objects related to the model, such as its
+                tokenizer.
+        """
+        import spacy
```

Comment:
We might want to just include spaCy as a global dependency, rather than an extra.